packages/edit-site/src/hooks/use-theme-style-variations/use-theme-style-variations-by-property.js

```diff
@@ -72,26 +70,26 @@ export function useCurrentMergeThemeStyleVariationsWithUserConfig( {
 	}, [] );
 	const { user: baseVariation } = useContext( GlobalStylesContext );
 
-	const variations = useMemo( () => {
-		return [
-			{
-				title: __( 'Default' ),
-				settings: {},
-				styles: {},
-			},
-			...variationsFromTheme,
-		];
-	}, [ variationsFromTheme ] );
+	return useMemo( () => {
+		const clonedBaseVariation = cloneDeep( baseVariation );
 
-	return useThemeStyleVariationsByProperty( {
-		variations,
-		property,
-		filter,
-		baseVariation: removePropertyFromObject(
-			cloneDeep( baseVariation ),
+		// Get user variation and remove the color settings.
+		const userVariation = removePropertyFromObject(
+			clonedBaseVariation,
 			property
-		),
-	} );
+		);
+		userVariation.title = __( 'Default' );
+
+		const variationsWithSinglePropertyAndBase = variationsFromTheme
+			.filter( ( variation ) => {
+				return isVariationWithSingleProperty( variation, property );
+			} )
+			.map( ( variation ) => {
+				return mergeBaseAndUserConfigs( baseVariation, variation );
```

Comment:
Should this be using the `clonedBaseVariation`? If `removePropertyFromObject` mutates the argument, then the [old code below](https://github.com/WordPress/gutenberg/pull/62469/files#diff-0ee3f0040b18ec7b1a93b7d799f26d84216f6b6a513a9a7337f1cc66c6d987ccL184) is merging the one without the property.

Reply:
Yes you're right, great catch. We want to apply only the typography styles from the variation, not include the ones from the base. I updated the PR and renamed a few variables too for clarity.